cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortServiceExceptionMapper.java

```diff
@@ -82,10 +82,11 @@ else if (ex instanceof IllegalArgumentException){
 				serviceErrorCode = Status.BAD_REQUEST.getStatusCode();
 				serviceErrorListCode = serviceErrorCode;
 			}
+			// FHIR Exception
 			else if (ex instanceof BaseServerResponseException) {
-				serviceErrorCode = ((BaseServerResponseException) ex).getStatusCode();
+				serviceErrorCode = Status.INTERNAL_SERVER_ERROR.getStatusCode();
 				serviceErrorListCode = serviceErrorCode;
-				description = ex.getMessage();
+				description = ((BaseServerResponseException) ex).getResponseBody();
```

Comment:
wasn't exactly sure either, as I would imagine that would depend on the underlying exception and not strictly defined in the java docs:
```java
	/**
	 * In a RESTful client, this method will be populated with the body of the HTTP respone if one was provided by the server, or <code>null</code> otherwise.
	 * <p>
	 * In a restful server, this method is currently ignored.
	 * </p>
	 */
	public String getResponseBody() {
		return myResponseBody;
	}
```

so I just did 5523463 in case.
Does that make sense to @tabishop?

Reply:
You should be able to force a response body with XML content by putting in a bad/garbage value for the fhir_data_source_id parameter on one of the REST api calls. If you have verified that the escaping ensures a proper json response is returned when you do that, I'm satisfied.

Reply:
I think Tim is right. I triggered a 500 Exception and got back the following response:
{
  "errors": [
    {
      "code": 500,
      "correlationId": "905e40ad-2728-4e54-8624-6c8ab3a8bc39",
      "description": "<OperationOutcome xmlns=\\\"http:\\/\\/hl7.org\\/fhir\\\">\\n  <id value=\\\"ac-1e-63-2a-a37ece52-e2c0-474d-8e2d-75c746e6e5de\\\"\\/>\\n  <issue>\\n    <severity value=\\\"fatal\\\"\\/>\\n    <code value=\\\"exception\\\"\\/>\\n    <details>\\n      <text value=\\\"FHIRPersistenceException: Unexpected exception while creating JDBC persistence layer: \\\"\\/>\\n    <\\/details>\\n  <\\/issue>\\n<\\/OperationOutcome>",
      "level": "ERROR",
      "message": "HTTP 500 Internal Server Error"
    }
  ],
  "statusCode": 500,
  "trace": "905e40ad-2728-4e54-8624-6c8ab3a8bc39"
}
The Json is messed up.